frontend/src/utils/genDuplicateSnippetName.js

```diff
@@ -2,19 +2,25 @@ import { SNIPPET_NAME_MAX_LENGTH } from './validationSchemas';
 
 const genDuplicateSnippetName = (fileName) => {
   const extensionIndex = fileName.lastIndexOf('.');
+  const copyTextIndex = fileName.lastIndexOf('-copy');
+  const dashIndex = fileName.lastIndexOf('_');
+  const existCopyNumber = Number(fileName.slice(dashIndex + 1));
+  console.log(copyTextIndex, dashIndex, existCopyNumber, extensionIndex);
 
   let finalName = '';
+  const extension = extensionIndex !== -1 ? fileName.slice(extensionIndex) : '';
 
-  if (extensionIndex === -1) {
-    finalName = `${fileName}-copy`;
-  } else {
-    const [name, extension] = [
-      fileName.slice(0, extensionIndex),
-      fileName.slice(extensionIndex + 1),
-    ];
-
-    finalName = `${name}-copy.${extension}`;
-  }
+  if (copyTextIndex !== -1) {
+    if (dashIndex > copyTextIndex) {
```

Comment:
if inside other if isnt a good practice. Maybe you can replace it with sequential checks?

Reply:
> if inside other if isnt a good practice. Maybe you can replace it with sequential checks?

Hi, Helen! 
Fixed nested conditions. Check, please.